Contents/Code/__init__.py

```diff
@@ -75,15 +75,28 @@ def Search(self, results, media, lang, manual, movie):
         # http://127.0.0.1:8111/api/serie/search?query=Clannad&level=1&apikey=d422dfd2-bdc3-4219-b3bb-08b85aa65579
 
         if movie:
-            filename = media.filename.split('%2F')[-1]
+            if media.filename:
+                filename = media.filename.split('%2F')[-1]
 
-            episode_data = HttpReq("api/ep/getbyfilename?filename=%s" % (filename))
-            movie_data = HttpReq("api/serie/fromep?id=%s" % (episode_data['id']))
-            
-            score = 100
-            title = movie_data['name'] + ' - ' + episode_data['name']
-            meta = MetadataSearchResult('%s' % (episode_data['id']), title, try_get(episode_data, 'year', None), score, lang)
-            results.Append(meta)
+                episode_data = HttpReq("api/ep/getbyfilename?filename=%s" % (filename))
+                movie_data = HttpReq("api/serie/fromep?id=%s" % (episode_data['id']))
+
+                score = 100 if movie_data['name'] == name else 85  # TODO: Improve this to respect synonyms./
+                title = movie_data['name'] + ' - ' + episode_data['name']
+                meta = MetadataSearchResult('%s' % (episode_data['id']), title, try_get(episode_data, 'year', None), score, lang)
+                results.Append(meta)
+
+            else: # For manual searches
+                prelimresults = HttpReq("api/serie/search?query=%s&level=%d&fuzzy=%d&ismovie=1" % (urllib.quote(name.encode('utf8')), 2, Prefs['Fuzzy']))
```

Comment:
According to swagger, setting `ismovie` is supposed to return only movies but it's not working.